### DISCUSSION TITLE
Cython installed in BEFORE ALL is not seen by `setup.py`

### DISCUSSION THREAD

**Discussion by abitrolly** [pypa/cibuildwheel#1146] [Q&A] [answered]
In this job https://github.com/abitrolly/opendht/runs/6950755175?check_suite_focus=true Cython is installed in BEFORE ALL step, but `setup.py` can not see it. Is this by design?

**Comment by Czaki** [comment, marked as answer]
yes. Because `BEFORE_ALL` is executed once on host python. To have option for cython visible in `setup.py` you need to install it in `BEFORE_BUILD` Then it will be installed in same python as used for build. 

But it may not work if build discover that isolated build is required. Then only option to hace cython is use `pyproject.toml` and put cython in `build-system` section in list `requires` like in this comment https://github.com/pypa/cibuildwheel/issues/1139#issuecomment-1157653804
